CI build failing after M.E.AI.Evaluations merging

```
CSC : error CS1566: Error reading resource 'Reporting.HTML.index.html' -- 'Could not find a part of the path 'D:\a\_work\1\s\src\Libraries\Microsoft.Extensions.AI.Evaluation.Reporting\TypeScript\html-report\dist\index.html'.' [D:\a\_work\1\s\src\Libraries\Microsoft.Extensions.AI.Evaluation.Reporting\CSharp\Microsoft.Extensions.AI.Evaluation.Reporting.csproj::TargetFramework=net8.0]
##[error]CSC(0,0): error CS1566: (NETCORE_ENGINEERING_TELEMETRY=Build) Error reading resource 'Reporting.HTML.index.html' -- 'Could not find a part of the path 'D:\a\_work\1\s\src\Libraries\Microsoft.Extensions.AI.Evaluation.Reporting\TypeScript\html-report\dist\index.html'.'
```
e.g. https://dev.azure.com/dnceng-public/public/_build/results?buildId=951692&view=logs&jobId=0bc77094-9fcd-5c38-f6e4-27d2ae131589&j=0bc77094-9fcd-5c38-f6e4-27d2ae131589&t=e2f678a7-b875-5c17-f40e-3e55bcefed52

cc: @peterwald, @shyamnamboodiripad

--- a/eng/pipelines/templates/BuildAndTest.yml
+++ b/eng/pipelines/templates/BuildAndTest.yml
@@ -43,7 +43,7 @@ steps:
             /bl:${{ parameters.repoLogPath }}/restore2.binlog
     displayName: Restore solution
 
-  # Temporarily pre-build the TypeScript byndle as a workaround for build failures in the MSBuild targets
+  # Temporarily pre-build the TypeScript bundle as a workaround to make the build more reliable.
   - pwsh: |
         cd $(Build.SourcesDirectory)/src/Libraries/Microsoft.Extensions.AI.Evaluation.Reporting/TypeScript
         npm install

--- a/eng/pipelines/templates/BuildAndTest.yml
+++ b/eng/pipelines/templates/BuildAndTest.yml
@@ -43,6 +43,7 @@ steps:
             /bl:${{ parameters.repoLogPath }}/restore2.binlog
     displayName: Restore solution
 
+  # Temporarily pre-build the TypeScript byndle as a workaround for build failures in the MSBuild targets
   - pwsh: |
         cd $(Build.SourcesDirectory)/src/Libraries/Microsoft.Extensions.AI.Evaluation.Reporting/TypeScript
         npm install
@@ -115,7 +116,6 @@ steps:
               -publish $(_PublishArgs)
               -configuration ${{ parameters.buildConfig }}
               /bl:${{ parameters.repoLogPath }}/transport.binlog
-              /p:EnableEvaluationReportBuild=true
               $(_OfficialBuildIdArgs)
       displayName: Build and publish docs transport package
 

--- a/eng/pipelines/templates/BuildAndTest.yml
+++ b/eng/pipelines/templates/BuildAndTest.yml
@@ -43,6 +43,13 @@ steps:
             /bl:${{ parameters.repoLogPath }}/restore2.binlog
     displayName: Restore solution
 
+  - pwsh: |
+        cd $(Build.SourcesDirectory)/src/Libraries/Microsoft.Extensions.AI.Evaluation.Reporting/TypeScript
+        npm install
+        node html-report/reset-devdata.js
+        npm run build
+    displayName: Pre-Build TypeScript Bundle
+
   - script: ${{ parameters.buildScript }}
             -build
             -configuration ${{ parameters.buildConfig }}
@@ -108,6 +115,7 @@ steps:
               -publish $(_PublishArgs)
               -configuration ${{ parameters.buildConfig }}
               /bl:${{ parameters.repoLogPath }}/transport.binlog
+              /p:EnableEvaluationReportBuild=true
               $(_OfficialBuildIdArgs)
       displayName: Build and publish docs transport package
 
@@ -119,5 +127,6 @@ steps:
               -warnAsError 1
               /bl:${{ parameters.repoLogPath }}/pack.binlog
               /p:Restore=false /p:Build=false
+              /p:EnableEvaluationReportBuild=true
               $(_OfficialBuildIdArgs)
       displayName: Pack, sign, and publish

